android/app/src/main/kotlin/com/jabook/app/jabook/audio/processors/NoiseGateAudioProcessor.kt

```
@@ -0,0 +1,305 @@
+        inputBuffers.clear()
+        outputBuffer = null
+        inputEnded = false
+    }
+
+    override fun reset() {
+        flush()
+        inputAudioFormat = null
+        isActive = false
+    }
+
+    private fun processBuffer(
```

detekt (1.23.8)

[warning] The function processBuffer appears to be too complex based on Cognitive Complexity (complexity: 25). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)